### PR TITLE
tuxguitar: fix start script

### LINKED ISSUE #227206: tuxguitar fails to start
### Describe the bug

Tuxguitar fails to start on `nixpkgs-unstable`, Gnome.

### Steps To Reproduce

```
$ tuxguitar
Invalid MIT-MAGIC-COOKIE-1 keyorg.eclipse.swt.SWTError: No more handles [gtk_init_check() failed]
        at org.eclipse.swt.SWT.error(SWT.java:4517)
        at org.eclipse.swt.widgets.Display.createDisplay(Display.java:908)
        at org.eclipse.swt.widgets.Display.create(Display.java:892)
        at org.eclipse.swt.graphics.Device.<init>(Device.java:156)
        at org.eclipse.swt.widgets.Display.<init>(Display.java:512)
        at org.eclipse.swt.widgets.Display.<init>(Display.java:503)
        at org.herac.tuxguitar.ui.swt.SWTApplication.createDisplay(SWTApplication.java:73)
        at org.herac.tuxguitar.ui.swt.SWTApplication.<init>(SWTApplication.java:18)
        at org.herac.tuxguitar.ui.swt.SWTApplicationFactory.createApplication(SWTApplicationFactory.java:9)
        at org.herac.tuxguitar.app.ui.TGApplication.<init>(TGApplication.java:27)
        at org.herac.tuxguitar.app.ui.TGApplication$1.createInstance(TGApplication.java:64)
        at org.herac.tuxguitar.app.ui.TGApplication$1.createInstance(TGApplication.java:62)
        at org.herac.tuxguitar.util.singleton.TGSingletonUtil.getInstance(TGSingletonUtil.java:12)
        at org.herac.tuxguitar.app.ui.TGApplication.getInstance(TGApplication.java:62)
        at org.herac.tuxguitar.app.synchronizer.TGSynchronizerControllerImpl.<init>(TGSynchronizerControllerImpl.java:18)
        at org.herac.tuxguitar.app.TuxGuitar.createUIContext(TuxGuitar.java:114)
        at org.herac.tuxguitar.app.TuxGuitar.createApplication(TuxGuitar.java:100)
        at org.herac.tuxguitar.app.TGMainSingleton.launchTuxGuitar(TGMainSingleton.java:57)
        at org.herac.tuxguitar.app.TGMainSingleton.launchSingleton(TGMainSingleton.java:45)
        at org.herac.tuxguitar.app.TGMainSingleton.main(TGMainSingleton.java:35)
```

### Notify maintainers

@ardumont 

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
this path will be fetched (0.01 MiB download, 0.04 MiB unpacked):
  /nix/store/viwmk7z6fxrd1xc09dximdk3w37kx1qn-stdenv-linux
copying path '/nix/store/viwmk7z6fxrd1xc09dximdk3w37kx1qn-stdenv-linux' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 6.2.10, NixOS, 23.05 (Stoat), 23.05.20230417.f00994e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by c31io**
1. Fix #227206: The package now uses `which java` in the startup script, so we need `which` as build input and `java` in `$PATH`.
2. Add `warpGAppsHook` to run in KDE.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/tu/tuxguitar/package.nix [hunk 1]
```diff
@@ -22,8 +24,13 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-FD1+7jV69E9AfTczjD6DOGD+pPlscg4o8A9ADBUM9B4=";
   };
 
+  buildInputs = [
+    which
+  ];
+
   nativeBuildInputs = [
     makeWrapper
+    wrapGAppsHook
```
Comment (getchoo) [inline pkgs/by-name/tu/tuxguitar/package.nix:33]:
```suggestion
    wrapGAppsHook3
```
https://github.com/NixOS/nixpkgs/pull/307077

File: pkgs/by-name/tu/tuxguitar/package.nix [hunk 2]
```diff
@@ -36,7 +43,7 @@ stdenv.mkDerivation (finalAttrs: {
     ln -s $out/share $out/bin/share
 
     wrapProgram $out/bin/tuxguitar \
-      --set JAVA "${jre}/bin/java" \
+      --set PATH "$PATH:${jre}/bin" \
```
Comment (getchoo) [inline pkgs/by-name/tu/tuxguitar/package.nix:46]:
```suggestion
      --prefix PATH : ${lib.makeBinPath [ jre which ]} \
      "${gappsWrapperArgs[@]}"
```
We can use `--prefix PATH` and `lib.makeBinPath` instead. We should also use the flags from `wrapGAppsHook3` here

File: pkgs/by-name/tu/tuxguitar/package.nix [hunk 3]
```diff
@@ -22,8 +24,13 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-FD1+7jV69E9AfTczjD6DOGD+pPlscg4o8A9ADBUM9B4=";
   };
 
+  buildInputs = [
+    which
+  ];
+
```
Comment (getchoo) [inline pkgs/by-name/tu/tuxguitar/package.nix:30]:
```suggestion
```

**Review by getchoo — CHANGES_REQUESTED** [review]
`dontWrapGApps = true` should also be set to avoid double wrapping the binary. See https://nixos.org/manual/nixpkgs/stable/#ssec-gnome-common-issues-double-wrapped

And welcome to Nixpkgs!

**Comment by c31io** [comment]
Thanks for the suggestions😊
Took me a while to figure out that I need to put warpProgram in postFixup to hook GSettings schemas.

File: pkgs/by-name/tu/tuxguitar/package.nix [hunk 4]
```diff
@@ -34,9 +39,12 @@ stdenv.mkDerivation (finalAttrs: {
     ln -s $out/dist $out/bin/dist
     ln -s $out/lib $out/bin/lib
     ln -s $out/share $out/bin/share
+  '';
 
+  postFixup = ''
     wrapProgram $out/bin/tuxguitar \
-      --set JAVA "${jre}/bin/java" \
+      "''${gappsWrapperArgs[@]}" \
+      --prefix PATH : ${lib.makeBinPath[ jre which ]} \
```
Comment (getchoo) [inline pkgs/by-name/tu/tuxguitar/package.nix:47]:
```suggestion
      --prefix PATH : ${lib.makeBinPath [ jre which ]} \
```
To make `nixfmt` happy

**Review by getchoo — CHANGES_REQUESTED** [review]

**Review by getchoo — APPROVED** [review]

**Pull request merged**